src/states/elder/mod.rs

```diff
@@ -384,35 +384,38 @@ impl Elder {
         self.send_direct_message(node.connection_info(), DirectMessage::ConnectionResponse);
     }
 
-    fn reset_parsec_with_data(&mut self, reset_data: ParsecResetData) -> Result<(), RoutingError> {
-        let drained_obs: Vec<_> = self
-            .parsec_map
-            .our_unpolled_observations()
-            .cloned()
-            .collect();
+    fn append_our_unpolled_observations(
```

Comment:
Why not keep this part of `reset_parsec_with_data`? Then you wouldn't have to call it explicitly in both `reset_parsec` and `finalise_split`.

Reply:
Because it also need to be in demote. Also later commit refactor it further.